lib/ios/BottomTabPresenter.m

```diff
@@ -1,68 +1,62 @@
 #import "BottomTabPresenter.h"
 #import "RNNTabBarItemCreator.h"
 #import "UIViewController+RNNOptions.h"
-#import "RNNDotIndicatorPresenter.h"
 #import "UIViewController+LayoutProtocol.h"
 
-@interface BottomTabPresenter ()
-@property(nonatomic, strong) RNNDotIndicatorPresenter* dotIndicatorPresenter;
-@end
-
 @implementation BottomTabPresenter
 
 - (instancetype)initWithDefaultOptions:(RNNNavigationOptions *)defaultOptions {
     self = [super init];
     self.defaultOptions = defaultOptions;
-    self.dotIndicatorPresenter = [[RNNDotIndicatorPresenter alloc] initWithDefaultOptions:defaultOptions];
     return self;
 }
 
-- (void)applyOptions:(RNNNavigationOptions *)options {
+- (void)applyOptions:(RNNNavigationOptions *)options child:(UIViewController *)child {
     RNNNavigationOptions * withDefault = [options withDefault:self.defaultOptions];
     
-    if (withDefault.bottomTab.badge.hasValue && [self.boundViewController.parentViewController isKindOfClass:[UITabBarController class]]) {
-        [self.boundViewController setTabBarItemBadge:withDefault.bottomTab.badge.get];
+    if (withDefault.bottomTab.badge.hasValue) {
+        [child setTabBarItemBadge:withDefault.bottomTab.badge.get];
     }
     
-    if (withDefault.bottomTab.badgeColor.hasValue && [self.boundViewController.parentViewController isKindOfClass:[UITabBarController class]]) {
-        [self.boundViewController setTabBarItemBadgeColor:withDefault.bottomTab.badgeColor.get];
+    if (withDefault.bottomTab.badgeColor.hasValue) {
+        [child setTabBarItemBadgeColor:withDefault.bottomTab.badgeColor.get];
     }
 }
 
-- (void)applyOptionsOnWillMoveToParentViewController:(RNNNavigationOptions *)options {
+- (void)applyOptionsOnWillMoveToParentViewController:(RNNNavigationOptions *)options  child:(UIViewController *)child {
     RNNNavigationOptions * withDefault = [options withDefault:self.defaultOptions];
     
+    if (withDefault.bottomTab.badge.hasValue) {
```

Comment:
Generally, `applyOptions` always sets a value - either an actual value defined by the user of the default value. Why isn't it the case here?